pkg/plugins/plugins.go

```diff
@@ -0,0 +1,86 @@
+package plugins
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+)
+
+var (
+	ErrNotImplements = errors.New("Plugin not implements particular driver")
+)
+
+type plugins struct {
+	sync.Mutex
+	plugins map[string]*Plugin
+}
+
+var storage = plugins{plugins: make(map[string]*Plugin)}
+
```

Comment:
While I was integrating with the plugin system earlier based on https://github.com/mavenugo/docker/commit/5153e3acc70066483bbffcfa0eed2b1c13d2fa92 , there was an useful construct `extpointHandlers` and a corresponding `plugins` api `Handle`. This construct provides a callback mechanism when a new plugin is loaded. I used that in my poc https://github.com/mavenugo/libnetwork/commit/b3dcb1fc398e6102fe06774abe9748ca6525e15a. Is there a reason why we removed this from the implementation ?

am NOT suggesting that we need that. am more interested to know the reasoning for not providing a handler callback.